inference-engine/tests/functional/plugin/myriad/ngraph/transformations/dynamic_to_static_shape_matmul.cpp

```diff
@@ -0,0 +1,413 @@
+        }
+        if (!matmul_setup.B.gather_idxs_for_transpose.empty()) {
+            const auto indices = ngraph::opset3::Constant::create(
+                    ngraph::element::i64, {matmul_setup.B.gather_idxs_for_transpose.size()}, matmul_setup.B.gather_idxs_for_transpose);
+            const auto axis = ngraph::opset3::Constant::create(ngraph::element::i64, {}, std::vector<int64_t>{0});
+            normalized_B_shape = std::make_shared<ngraph::opset3::Gather>(normalized_B_shape, indices, axis);
+        }
+
+        // Common
+        ngraph::OutputVector output_dims;
+        if (!matmul_setup.A.batch_gather_idxs.empty()) {
+            const auto max_shape = std::make_shared<ngraph::opset3::Maximum>(normalized_A_shape, normalized_B_shape);
+            const auto indices = ngraph::opset3::Constant::create(
+                    ngraph::element::i64, {matmul_setup.A.batch_gather_idxs.size()}, matmul_setup.A.batch_gather_idxs);
+            const auto axis = ngraph::opset3::Constant::create(ngraph::element::i64, {}, std::vector<int64_t>{0});
+            const auto batch_dims = std::make_shared<ngraph::opset3::Gather>(max_shape, indices, axis);
+            output_dims.push_back(batch_dims);
+        }
+
+        const auto input_channels = std::make_shared<ngraph::opset3::Gather>(
+                normalized_A_shape,
+                ngraph::opset3::Constant::create(ngraph::element::i64, {1}, {matmul_setup.A.channel_idx}),
+                ngraph::opset3::Constant::create(ngraph::element::i64, {}, std::vector<int64_t>{0}));
+        const auto output_channels = std::make_shared<ngraph::opset3::Gather>(
+                normalized_B_shape,
+                ngraph::opset3::Constant::create(ngraph::element::i64, {1}, {matmul_setup.B.channel_idx}),
+                ngraph::opset3::Constant::create(ngraph::element::i64, {}, std::vector<int64_t>{0}));
+        output_dims.push_back(input_channels);
+        output_dims.push_back(output_channels);
+
+        const auto output_shape = std::make_shared<ngraph::opset3::Concat>(output_dims, 0);
+        const auto dsr_final = std::make_shared<ngraph::vpu::op::DynamicShapeResolver>(node, output_shape);
+
+        const auto function = std::make_shared<ngraph::Function>(
+                ngraph::NodeVector{dsr_final},
+                ngraph::ParameterVector{input_A, input_B, input_B_dsr},
+                "Transformed-MatMul");
+
+        return function;
+    }
+};
+
+
+TEST_P(DynamicToStaticShapeMatMul, CompareFunctions) {
+}
+INSTANTIATE_TEST_CASE_P(MatMul, DynamicToStaticShapeMatMul, combinations);
+
+TEST_P(DynamicToStaticShapeMatMulAInputDSR, CompareFunctions) {
+}
+INSTANTIATE_TEST_CASE_P(MatMulSingleDSR, DynamicToStaticShapeMatMulAInputDSR, combinations);
+
+TEST_P(DynamicToStaticShapeMatMulBInputDSR, CompareFunctions) {
+}
+INSTANTIATE_TEST_CASE_P(MatMulSingleDSR, DynamicToStaticShapeMatMulBInputDSR, combinations);
```

Comment:
What is the difference between 3 test classes? Now it looks like the only difference is in input nodes generation - processing is the same. If so, I think we can get rid of code duplication

Reply:
Agree. I redesigned testing approach for MatMul. Please see if it is what you meant.